pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -264,11 +265,13 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 	provisionedPhase := "Provisioned"
 	// runningPhase is the status of the machine when it is in the `Running` state, indicating that it is configured into a node
 	runningPhase := "Running"
+	// runningPhase is the status of the machine when it is in the `Deleting` state
+	deletingPhase := "Deleting"
 	if machine.Status.Phase == nil {
 		// Phase is nil and should be ignored by WMCO until phase is set
 		// TODO: Instead of requeuing ignore certain events: https://issues.redhat.com/browse/WINC-500
 		return reconcile.Result{}, fmt.Errorf("could not get the phase associated with machine %s", machine.Name)
-	} else if *machine.Status.Phase == runningPhase {
+	} else if *machine.Status.Phase == runningPhase || *machine.Status.Phase == deletingPhase {
```

Comment:
why is the deleting phase check and what are we changing with the upgrade here?

Reply:
Changed the predicate to get Delete events. We need delete events, so that we could remove the deleted node entry in the Endpoints object

Reply:
This is a big change. Please call this out in the commit message mentioning why you had to make the change.